File: api/app/Models/SubscriptionPlan.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class SubscriptionPlan extends Model
{
    use HasFactory;

    protected $fillable = [
        'name',
        'slug',
        'description',
        'price',
        'duration_days',
        'max_stores',
        'max_products',
        'max_portfolio',
        'can_use_sponsored_zones',
        'max_sections',
        'is_active',
        'sort_order',
    ];

    protected $casts = [
        'price' => 'decimal:2',
        'is_active' => 'boolean',
        'can_use_sponsored_zones' => 'boolean',
    ];

    /**
     * Get all subscriptions using this plan.
     */
    public function subscriptions()
    {
        return $this->hasMany(Subscription::class, 'plan_id');
    }

    /**
     * Scope for active plans only.
     */
    public function scopeActive($query)
    {
        return $query->where('is_active', true);
    }

    /**
     * Check if this is the free trial plan.
     */
    public function isFreeTrial(): bool
    {
        return $this->slug === 'free';
    }

    /**
     * Check if products are unlimited.
     */
    public function hasUnlimitedProducts(): bool
    {
        return $this->max_products === -1;
    }

    /**
     * Get formatted price.
     */
    public function getFormattedPriceAttribute(): string
    {
        if ($this->price == 0) {
            return 'Free';
        }
        return number_format((float) $this->price, 0) . ' DA';
    }

    /**
     * Get duration text.
     */
    public function getDurationTextAttribute(): string
    {
        if ($this->duration_days <= 30) {
            return $this->duration_days . ' days';
        }
        $months = intval($this->duration_days / 30);
        if ($months >= 12) {
            $years = intval($months / 12);
            return $years . ' year' . ($years > 1 ? 's' : '');
        }
        return $months . ' months';
    }
}

File: lib/models/subscription_plan.dart
class SubscriptionPlan {
  final int id;
  final String name;
  final String slug;
  final String? description;
  final double price;
  final int durationDays;
  final int maxStores;
  final int maxProducts;
  final int maxPortfolio;
  final bool canUseSponsoredZones;
  final int maxSections;
  final bool isActive;
  final int sortOrder;

  SubscriptionPlan({
    required this.id,
    required this.name,
    required this.slug,
    this.description,
    required this.price,
    required this.durationDays,
    required this.maxStores,
    required this.maxProducts,
    required this.maxPortfolio,
    this.canUseSponsoredZones = false,
    this.maxSections = 5,
    this.isActive = true,
    this.sortOrder = 0,
  });

  factory SubscriptionPlan.fromJson(Map<String, dynamic> json) {
    return SubscriptionPlan(
      id: json['id'] as int,
      name: json['name'] as String,
      slug: json['slug'] as String,
      description: json['description'] as String?,
      price: (json['price'] is String)
          ? double.parse(json['price'])
          : (json['price'] as num).toDouble(),
      durationDays: json['duration_days'] as int,
      maxStores: json['max_stores'] as int,
      maxProducts: json['max_products'] as int,
      maxPortfolio: json['max_portfolio'] as int,
      canUseSponsoredZones: json['can_use_sponsored_zones'] as bool? ?? false,
      maxSections: json['max_sections'] as int? ?? 5,
      isActive: json['is_active'] as bool? ?? true,
      sortOrder: json['sort_order'] as int? ?? 0,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'slug': slug,
      'description': description,
      'price': price,
      'duration_days': durationDays,
      'max_stores': maxStores,
      'max_products': maxProducts,
      'max_portfolio': maxPortfolio,
      'is_active': isActive,
      'sort_order': sortOrder,
    };
  }

  /// Check if this is the free trial plan
  bool get isFreeTrial => slug == 'free';

  /// Check if products are unlimited
  bool get hasUnlimitedProducts => maxProducts == -1;

  /// Get formatted price
  String get formattedPrice {
    if (price == 0) return 'Free';
    return '${price.toInt()} DA';
  }

  /// Get duration text
  String get durationText {
    if (durationDays <= 30) {
      return '$durationDays days';
    }
    final months = durationDays ~/ 30;
    if (months >= 12) {
      final years = months ~/ 12;
      return '$years year${years > 1 ? 's' : ''}';
    }
    return '$months months';
  }

  /// Get badge color based on plan
  int get badgeColor {
    switch (slug) {
      case 'gold':
        return 0xFFFFD700; // Gold
      case 'silver':
        return 0xFFC0C0C0; // Silver
      default:
        return 0xFF10B981; // Green for free
    }
  }
}
